qiskit/visualization/pulse_v2/drawing_objects.py

```diff
@@ -57,23 +69,26 @@
 due to lack of the effective unittest.
 """
 from abc import ABC, abstractmethod
-from typing import Dict, Any, Optional, Union
+from typing import Dict, Any, Optional, Union, List
 
 import numpy as np
 
 from qiskit.pulse import channels
-from qiskit.visualization.exceptions import VisualizationError
+from qiskit.visualization.pulse_v2 import types
 
 
 class ElementaryData(ABC):
     """Base class of the pulse visualization interface."""
+    __hash__ = None
```

Comment:
Hash is already implicitly set to `None` as you have overridden `__eq__` but not `__hash__`? https://docs.python.org/3/reference/datamodel.html#object.__hash__

Reply:
Yeah this is for making it explicit. In interactive plotter we need to manage data source (`drawing_object`) and associated visualization data to dynamically update the image. However `drawing_object` is also mutable and cannot be a dict key. So this recommend to use `.data_key` method which is insensitive to update of `drawing_object`.